nifi-nar-bundles/nifi-adx-bundle/nifi-adx/src/main/java/org/sebkenter/nifi/adx/AzureAdxConnectionService.java

```diff
@@ -0,0 +1,167 @@
+            .description("Azure application ID Key for accessing the ADX-Cluster")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor APP_TENANT = new PropertyDescriptor
+            .Builder().name("APP_TENANT")
+            .displayName("Application Tenant")
+            .description("Azure application tenant for accessing the ADX-Cluster")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    private static final List<PropertyDescriptor> properties;
+
+    static {
+        final List<PropertyDescriptor> props = new ArrayList<>();
+        props.add(INGEST_URL);
+        props.add(APP_ID);
+        props.add(APP_KEY);
+        props.add(APP_TENANT);
+        properties = Collections.unmodifiableList(props);
+    }
+
+    private IngestClient _ingestClient;
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    /**
+     * @param context
+     *            the configuration context
+     * @throws InitializationException
+     *             if unable to create a database connection
+     */
+    @OnEnabled
+    public void onEnabled(final ConfigurationContext context) throws InitializationException {
+        ComponentLog log = getLogger();
+
+        log.info("Starting Azure ADX Connection Service...");
+
+        final String ingestUrl = context.getProperty(INGEST_URL).evaluateAttributeExpressions().getValue();
+        final String app_id = context.getProperty(APP_ID).evaluateAttributeExpressions().getValue();
+        final String app_key = context.getProperty(APP_KEY).evaluateAttributeExpressions().getValue();
+        final String app_tenant = context.getProperty(APP_TENANT).evaluateAttributeExpressions().getValue();
+
+        if(this._ingestClient != null) {
+            onStopped();
+        }
+
+        this._ingestClient = createAdxClient(ingestUrl, app_id, app_key, app_tenant);
+    }
+
+    @OnStopped
+    public final void onStopped() {
+        if (this._ingestClient != null) {
+            try {
+                this._ingestClient.close();
+            } catch(IOException e) {
+                getLogger().error("Closing Azure ADX Client failed with: " + e.getMessage(), e);
+            } finally {
+                this._ingestClient = null;
+            }
+        }
+    }
+
+    @OnDisabled
+    public void shutdown() {
+
+    }
+
+    @Override
+    public void execute() throws ProcessException {
+
+    }
+
+    protected IngestClient createAdxClient(String ingestUrl, String appId, String appKey, String appTenant)
+    {
+        IngestClient client;
+
+        try {
+            ConnectionStringBuilder csb =
+                    ConnectionStringBuilder.createWithAadApplicationCredentials(ingestUrl, appId, appKey, appTenant);
+
+            try {
```

Comment:
do we need a nested try clause?

Reply:
Perhaps we don't need that. I took it from the azure-kusto-java sample for file ingestions: https://github.com/Azure/azure-kusto-java/blob/master/samples/src/main/java/FileIngestion.java